### ISSUE TITLE
Fully optional inputs should not require a 2nd parameter

### ISSUE THREAD

**Issue by NuroDev** [total-typescript/untypeable#3]
### What is the issue?

Currently if you pass in an input where all properties are optional you are still required to pass in an empty object as a second argument. For example:

```ts
import { initUntypeable, createTypeLevelClient } from "untypeable";

const u = initUntypeable();

interface User {
  id: string;
  name: string;
};

const router = u.router({
  "/user": u.input<{ id?: string }>().output<User>(),
});

const client = createTypeLevelClient<typeof router>((path, input) => {
  return fetch(`http://localhost:3000${path}?${new URLSearchParams(input)}`).then(
    (res) => res.json(),
  );
});

const user = await client("/user", {});
//                                  ^ Without this empty object TypeScript will complain that a 2nd argument is missing
```

Ideally it would be better if it was checked it all properties for an input record are optional & if so make the 2nd argument for the client function optional / default to an empty object.

**Comment by mattpocock** [comment]
Fixed on main! Will release a new version soon.

**Comment by mattpocock** [comment]
Published in 0.2.1
